# Patch for bug #36689 (TextArea onChange failure)
# Current way to insert characters into a text/input area resets the previous value to current one
# everytime a new char is inserted. So, when the widget loses focus the previous value will be always 
# the same of the current one and onchange event will never be dispatched.
# 
# This patch only comments the function which resets the previous value on a character insertion.
Index: mozilla/layout/forms/nsTextControlFrame.cpp
===================================================================
--- mozilla.orig/layout/forms/nsTextControlFrame.cpp
+++ mozilla/layout/forms/nsTextControlFrame.cpp
@@ -2772,17 +2772,17 @@
       }
 
       NS_ENSURE_STATE(weakFrame.IsAlive());
       if (outerTransaction)
         mNotifyOnInput = PR_TRUE;
 
       if (focusValueInit) {
         // Reset mFocusedValue so the onchange event doesn't fire incorrectly.
-        InitFocusedValue();
+        // InitFocusedValue();
       }
     }
 
     NS_ENSURE_STATE(weakFrame.IsAlive());
     nsIScrollableView* scrollableView = GetScrollableView();
     if (scrollableView)
     {
       // Scroll the upper left corner of the text control's
